### ISSUE TITLE
Error messge still active although get_webDir() in config.php was changed

### ISSUE THREAD

**Issue by efelon** [dchang0/torrentwatch-xa#1]
Hi,

**Setup**:
Raspbian stretch
Nginx 1.10.3 (WebDir /var/www/torrentwatch)
```
location /torrentwatch {
	index index.html;

	location ~ \.php$ {
		include snippets/fastcgi-php.conf;
		fastcgi_pass unix:/var/run/php/php7.0-fpm.sock;
	}
    }
```
PHP 7.0 (fpm)

**Logs**

I was not able to find any hints in nginx error logs nor in _/tmp/twxalog_.

**Description**

When I change the location to the default one 

> /var/www/html/torrentwatch-xa

 the error disappears. It also doesn't show up, even if I change the config.php get_webDir() to return another folder.

Other than that i can't find any other issue at the moment (tested now only for ~30 minutes). I'm able to connect to a transmission instance on another machine and add a torrent.

Keep up the resurrection ;)
Regards,
Martin

**Comment by dchang0** [comment]
I went ahead and changed the root /var/www/html to root /var/www so that I could put torrentwatch-xa's web portion in /var/www/torrentwatch.

Then I edited the values in /var/lib/torrentwatch/config.php in the getWebDir() function and uncommented the function.

I DO get PHP errors in /var/log/nginx/error.log like this:

```
2018/04/28 17:58:18 [error] 22907#22907: *14 FastCGI sent in stderr: "PHP message: PHP Warning:  require_once(twxa_tools.php): failed to open stream: No such file or directory in /var/www/torrentwatch/torrentwatch-xa.php on line 20
PHP message: PHP Fatal error:  require_once(): Failed opening required 'twxa_tools.php' (include_path='.:/usr/share/php:/var/lib/torrentwatch-xa/lib') in /var/www/torrentwatch/torrentwatch-xa.php on line 20"
```

You would be able to change the paths IF you used a Find and Replace through all the source code and replaced /var/www/html/torrentwatch-xa with /var/www/torrentwatch wherever it is found.

It's pretty obvious I have to redesign the way torrentwatch-xa refers to the paths. I probably have to move the getWebDir() and getBaseDir() functions into torrentwatch-xa.php so that they are the first function called. It makes more sense logically to put them in config.php, but that doesn't work practically.

I will leave this ticket open for now; it may be a while before I make that big a change (probably the next major release, which would be 1.0).

**Comment by efelon** [comment]
Hi dchang0,

you're way to fast for me today. I will answer anyway. I did indeed change the root to /var/www. I just like to place my web-folders outside of the default nginx /var/www/html. Without any other reason. :)

Anyway, below you see my nginx config for the server. I left the original "root" and "index" as comment, so you can see what i was trying. Then i included roots for every location:

```
server {
	listen 80 default_server;
	listen [::]:80 default_server;

	access_log off;

	#root /var/www;

	# Add index.php to the list if you are using PHP
	#index index.php index.html index.htm;

	server_name _;

	auth_basic           "Only one shall pass";
        auth_basic_user_file /etc/apache2/.htpasswd;

	location / {
                root /var/www/aria2;
		index index.html;
		try_files $uri $uri/ =404;
	}
	
	location /torrentwatch-xa {
		root /var/www/html;
		index index.html;
		try_files $uri $uri/ =404;
	
		# pass PHP scripts to FastCGI server
		#
		location ~ \.php$ {
		include snippets/fastcgi-php.conf;
		fastcgi_pass unix:/var/run/php/php7.0-fpm.sock;
       		fastcgi_read_timeout 300;
        	fastcgi_request_buffering off;
		}
	}

	location ~ /\.ht {
		deny all;
	}
}
```

At the moment it is doing it's job for a few days now without major problems. :+1:  I just found a small GUI problem, where matches in the header of a feed list "(n)" display 0 when opening the tab with collapsed feed lists:

![feed_lists_collapsed](https://user-images.githubusercontent.com/10722552/39400016-78d4b572-4b28-11e8-8c0f-16493e9b7320.jpg)

After extending the list with matches once and collapsing it again it shows them correctly.

Thanks for your time and investigations,
Martin

**Comment by dchang0** [comment]
I fixed the Feed header item counter. You only need the new Javascript file (and to refresh your browser cache).

The file is var/www/html/torrentwatch-xa/javascript/torrentwatch-xa.js

I did not increment the version. It is still 0.9.0 for this tiny bugfix.

It will likely be the next big release before I fix the getWebDir() and getBaseDir() bug, so this issue will remain open until then.

**Comment by dchang0** [comment]
I have fixed the get_webDir() and get_baseDir() bug in 1.0.0, to be released in a month or two.
I am successfully testing it on a Raspberry Pi 3 Model B running the latest Raspbian Lite Stretch and nginx instead of Apache2. The web directory I am currently testing is /var/www/torrentwatch and it only needs to be set in two places: inside config.php in the get_webDir() function and in torrentwatch-xa-cron

**Comment by dchang0** [comment]
Both bugs have been fixed now with the release of 1.0.0.

Please note that changing the web and base directories is done in only one file: config.php, but the setting for the base directory is cached in the torrentwatch-xa.config file and may need to be changed there too if you are carrying over a config file from a previous installation.
